### ISSUE TITLE
Local cache?

### ISSUE THREAD

**Issue by rstefko** [BlitterStudio/MB_SubSonic#65]
In [v2.30](https://github.com/midwan/MB_SubSonic/releases/tag/v2.30) there is: Removed pre-caching function. Instead, only load folders on-demand and let MusicBee do any local library caching itself.

How is the MusicBee supposed to do the caching? I am clicking through the Subsonic folders / play songs, but I don't see any list of all songs in MusicBee. The Music tab is still empty.

**Comment by rstefko** [comment]
When I select songs from Subsonic and righ-click them there is option Send to -> Library. It looks like to be working, but not sure this is how it is supposed to work (a lot of manual work).

**Comment by midwan** [comment]
You should be seeing files/folders when browsing under the Subsonic tree, if the connection to the server is working.
What are you running from the server side?
What kind of authentication did you use?

**Comment by rstefko** [comment]
Yes that is OK. But I would like to "Play Shuffled" through all songs in the library. But I can only play items from one folder / node. There is no list of all songs. So I thought that in previous versions the precaching loaded the songs to the library.

**Comment by midwan** [comment]
Ah then I misunderstood what you're trying to do.
Yes, the pre-caching would iterate through all the subfolders and get all songs contained in them, but it was quite slow with huge music libraries, and there were several issues that were caused by the process which lead me to just take the whole thing away.

I'll check if there's another/better way to do what you're looking for.

**Comment by midwan** [comment]
Although local cache is not back, the changes in the latest version might make the experience better for you.
Selecting the root node will now get you 100 random songs from the library.
Selecting any sub-node, will recursively retrieve and show all songs (this might take a while, depending on the structure and size of your library).
Also, added browsing by tags.

The new version will be released soon, need to do some more extensive testing.
